09_unique.rb

```diff
@@ -17,7 +17,37 @@
 # Check your solution by running the tests:
 # ruby tests/09_unique_test.rb
 #
+#
+#array style
+def unique (mylist)
+clean = []
+duplicate = []
+  mylist.each do |value|
+    if !clean.include?(value)
+        clean.push(value)
+    else
+      duplicate.push(value)
+    end
+  end
+  return clean
+end
+
+#hash style
+def unique (mylist)
+  clean = {}
+  duplicate = []
+  mylist.each do |value|
+    if value != true
```

Comment:
I think this condition runs _every_ time, since you're setting the value to `value` not `true`.

This code should still work if you remove lines 40 & 42-44